integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/spark/SparkOnKubernetesTestsSuite.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.kubernetes.test.spark
+
+import org.apache.kyuubi.{Logging, WithKyuubiServer}
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.kubernetes.test.MiniKube
+import org.apache.kyuubi.operation.SparkQueryTests
+
+// TODO Support Spark Cluster mode
```

Comment:
That's a tricky thing, got asked from user several times. The official Spark docker image has only one UID 185 w/o user name[1], Kyuubi will append `--proxy-user` to `spark-submit` command if `--keytab` absent, then cause `185 is not allowed to impersonate <user>`. Do you encounter other blockers for Cluster mode?
[1]https://spark.apache.org/docs/latest/running-on-kubernetes.html#user-identity

Reply:
Actually I have not tried test in cluster mode, it requires a remote file system. which is quite different with client mode

Reply:
Maybe we could do a post in the discussion and ask if everyone have any better ideas.

Reply:
Another thing to pay attention to, with cluster mode we need found somewhere to store `kyuubi-spark-engine jar`. Specify `spark.kubernetes.file.upload.path` or put it into docker image.

Reply:
@hddong Any way to tell Kyuubi its in the Image?